src/zarr/core/array_spec.py

```diff
@@ -5,40 +5,68 @@
 
 import numpy as np
 
-from zarr.core.common import MemoryOrder, parse_fill_value, parse_order, parse_shapelike
+from zarr.core.common import (
+    MemoryOrder,
+    parse_fill_value,
+    parse_order,
+    parse_shapelike,
+    parse_write_empty_chunks,
+)
+from zarr.core.config import config as zarr_config
 
 if TYPE_CHECKING:
     from zarr.core.buffer import BufferPrototype
     from zarr.core.common import ChunkCoords
 
 
+@dataclass(frozen=True)
+class ArrayConfig:
+    order: MemoryOrder
+    write_empty_chunks: bool
+
+    def __init__(
```

Comment:
this design prevents us from having `None` as a valid value for a configuration parameter, because it will not possible to distinguish "parameter was unset" from "parameter was set to `None`". A solution is to use a classmethod that takes a dictionary where some keys may not be present, which unambiguously express a missing parameter. I am working on this.